src/ng-select/ng-select.component.ts

```diff
@@ -187,7 +187,9 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         if (changes.multiple) {
             this.itemsList.clearSelected();
         }
-        if (changes.items) {
+        if (changes.items && 
+            (changes.items.firstChange || 
+            (!changes.items.firstChange && changes.items.currentValue.length !== changes.items.previousValue.length))) {
```

Comment:
What happens in this case

current
```
items = ['d', 'e', 'cf'];
```
previous
```
items = ['a', 'b', 'c'];
```
Both have the same length but are completely different.
I'm not sure what performance problems do you have?